src/mappings/erc20Templates.ts

```diff
@@ -204,6 +198,35 @@ export function handleNewPaymentCollector(event: NewPaymentCollector): void {
   token.save()
 }
 
+export function handleProviderFee(event: ProviderFee): void {
+  const providerFee: string = `{"providerFeeAddress": "${event.params.providerFeeAddress.toHex()}", "providerFeeToken": "${event.params.providerFeeToken.toHex()}", "providerFeeAmount": "${
+    event.params.providerFeeAmount
+  }", "providerData": "${event.params.providerData.toHexString()}", "v": "${
+    event.params.v
+  }", "r": "${event.params.r.toHexString()}", "s": "${event.params.s.toHexString()}", "validUntil": "${
+    event.params.validUntil
+  }"}`
+
+  const orderId = getOrderId(
+    event.transaction.hash.toHex(),
+    event.address.toHex(),
+    event.transaction.from.toHex()
+  )
+  const order = Order.load(orderId)
+  let orderReuse = OrderReuse.load(event.transaction.hash.toHex())
+  if (order) {
+    order.providerFee = providerFee
+    order.save()
```

Comment:
probably can just return here, early escape so next statement wont have to be in any condition closure

Reply:
true